app/controllers/items_controller.rb

```diff
@@ -19,6 +19,16 @@ def update
     render :edit
   end
 
+  def reserve
+    item = Item.find(params[:item_id])
+    wishlist = Wishlist.find(params[:wishlist_id])
+    authorize! :reserve, item
+    Item.change_reserved(item, params[:item][:is_reserved])
+    redirect_to wishlists_for_guest_path(wishlist.token), notice: t(".success")
+  rescue CanCan::AccessDenied
+    redirect_to wishlists_for_guest_path(wishlist.token), alert: t(".error")
```

Comment:
Huh, OK. I mean, I wouldn't enforce this on the user, his list, he can do whatever he wants to with it. But OK.